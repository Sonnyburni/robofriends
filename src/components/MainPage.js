import'./MainPage.css';
import React, { useState, useEffect } from 'react';
import CardList from '../components/CardList';
import SearchBox from '../components/SearchBox';
import Scroll from '../components/Scroll';
import ErrorBoundary from '../components/ErrorBoundary';
import Header from	'../components/header';
import { useOnlineStatus } from '../components/online';


const MainPage = ({onSearchChange, pending, text, robots}) => {

	const online = useOnlineStatus();
    const [searchResults, setSearchResults] = useState([]);

    useEffect(() => {
        let filteredRobots = robots.filter(robot => {
            return(
                robot.name.toLowerCase().includes(text.toLowerCase())
            );
        });
        setSearchResults(filteredRobots);
    }, [text,robots])

    const newRobot = searchResults;

	return(
		 online ?
			<div className='tc'>
				<Scroll>
					<Header />
					<SearchBox searchChange ={onSearchChange}/>
				</Scroll>
				    { pending ? <h1>Loading...</h1> :
					<ErrorBoundary>
					 {
                    text === "" ? <CardList robots={ robots }/> : <CardList robots={ newRobot }/>
                     }
					</ErrorBoundary>	
					}			
				</div>	
				:<h1>Please check your internet connection</h1>
			);
		}
		
	

export default MainPage;