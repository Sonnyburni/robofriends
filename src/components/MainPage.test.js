import {render, screen, fireEvent} from '@testing-library/react';
import ShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';
import MainPage from './MainPage';
import userEvent from '@testing-library/user-event';



const renderer = new ShallowRenderer();

beforeEach(() => {
})

it('renders MainPage without crashing', () => {
    const onSearchChange = jest.fn();
const pending = false;
const text = 'a';
const robots = [];
    renderer.render(<MainPage onSearchChange = {onSearchChange}  pending = {pending} text = {text} robots={robots} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot()
});

it('renders robots correctly', () => {
    const onSearchChange = jest.fn();
    const pending = false;
    const text = 'o';
    const robots = [{
        id:3,
        name:'John',
        email:'john@gmail.com'
    }];
    const user = userEvent.setup()
    render(<MainPage onSearchChange = {onSearchChange}  pending = {pending} text = {text} robots={robots} />);
    screen.getByText('John');
})