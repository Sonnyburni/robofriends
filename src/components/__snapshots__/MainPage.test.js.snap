// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainPage without crashing 1`] = `
<div
  className="tc"
>
  <Scroll>
    <Header />
    <SearchBox
      searchChange={[MockFunction]}
    />
  </Scroll>
  <ErrorBoundary>
    <CardList
      robots={Array []}
    />
  </ErrorBoundary>
</div>
`;
